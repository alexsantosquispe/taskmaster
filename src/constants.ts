import { PRIORITY_TYPES } from './models/types';

export const THEME_KEY = 'theme';

export const PRIORITY_COLORS = {
  [PRIORITY_TYPES.LOW]:
    'bg-amber-100 text-yellow-800 border border-yellow-800 dark:bg-yellow-600/30 dark:border-yellow-400 dark:text-yellow-400',
  [PRIORITY_TYPES.MEDIUM]:
    'bg-orange-100 text-amber-800 border border-amber-800 dark:bg-orange-600/20 dark:border-orange-400 dark:text-orange-400',
  [PRIORITY_TYPES.HIGH]:
    'bg-red-100 text-rose-700 border border-rose-700 dark:bg-red-600/20 dark:border-red-400 dark:text-red-400'
};

export const currentUser = {
  name: 'Alex Santos',
  email: 'alex.santos456@gmail.com',
  avatarUrl: '/img/avatar.webp'
};
