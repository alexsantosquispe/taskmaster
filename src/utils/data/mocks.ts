import {
  PRIORITY_TYPES,
  STATUS_TYPES,
  type TaskType
} from '../../models/types';

export const BACKLOG_TASKS: TaskType[] = [
  {
    id: '1',
    title: 'Set up analytics tools',
    description:
      'Install and configure Google Analytics and Hotjar to monitor user behavior and traffic.',
    status: STATUS_TYPES.BACKLOG,
    createdAt: '2025-07-01T10:24:00-04:00',
    lastUpdate: '2025-07-05T14:13:00-04:00',
    priority: PRIORITY_TYPES.MEDIUM,
    progress: 0
  },
  {
    id: '2',
    title: 'Gather user feedback on onboarding',
    description:
      'Collect and analyze user feedback from early access users about the onboarding experience.',
    status: STATUS_TYPES.BACKLOG,
    createdAt: '2025-07-02T11:30:00-04:00',
    lastUpdate: '2025-07-06T09:00:00-04:00',
    priority: PRIORITY_TYPES.HIGH,
    progress: 0
  },
  {
    id: '18',
    title: 'Write project README',
    description:
      'Added complete project documentation including setup, scripts, and folder structure.',
    status: STATUS_TYPES.DONE,
    createdAt: '2025-07-03T13:40:00-04:00',
    lastUpdate: '2025-07-06T10:00:00-04:00',
    priority: PRIORITY_TYPES.LOW,
    progress: 100
  }
];

export const ON_HOLD_TASKS: TaskType[] = [
  {
    id: '7',
    title: 'Finalize pricing model options',
    description:
      'Pricing structure needs stakeholder review before publishing. Waiting for finance team input.',
    status: STATUS_TYPES.ON_HOLD,
    createdAt: '2025-07-03T10:00:00-04:00',
    lastUpdate: '2025-07-09T11:45:00-04:00',
    priority: PRIORITY_TYPES.LOW,
    progress: 75
  }
];

export const IN_PROGRESS_TASKS: TaskType[] = [
  {
    id: '12',
    title: 'Implement authentication flow',
    description:
      'OAuth integration is partially done; currently wiring up token refresh and user session handling.',
    status: STATUS_TYPES.IN_PROGRESS,
    createdAt: '2025-07-06T09:30:00-04:00',
    lastUpdate: '2025-07-10T10:45:00-04:00',
    priority: PRIORITY_TYPES.HIGH,
    progress: 45
  },
  {
    id: '13',
    title: 'Integrate Supabase for data storage',
    description:
      'Connection established. Working on CRUD operations and testing permissions for different roles.',
    status: STATUS_TYPES.IN_PROGRESS,
    createdAt: '2025-07-08T14:00:00-04:00',
    lastUpdate: '2025-07-10T09:30:00-04:00',
    priority: PRIORITY_TYPES.HIGH,
    progress: 80
  }
];

export const REVIEW_TASKS: TaskType[] = [
  {
    id: '8',
    title: 'Update brand guidelines',
    description:
      'Rebranding paused until feedback from the design lead is received.',
    status: STATUS_TYPES.REVIEW,
    createdAt: '2025-07-02T14:20:00-04:00',
    lastUpdate: '2025-07-06T17:00:00-04:00',
    priority: PRIORITY_TYPES.MEDIUM,
    progress: 100
  },
  {
    id: '9',
    title: 'Legal review for terms of service',
    description:
      'Sent draft to legal team. Awaiting confirmation before moving forward.',
    status: STATUS_TYPES.REVIEW,
    createdAt: '2025-07-05T13:10:00-04:00',
    lastUpdate: '2025-07-08T12:30:00-04:00',
    priority: PRIORITY_TYPES.HIGH,
    progress: 100
  },
  {
    id: '19',
    title: 'Create logo and favicon',
    description:
      'Exported multiple sizes and formats for the logo. Favicon added to HTML head.',
    status: STATUS_TYPES.DONE,
    createdAt: '2025-07-04T08:00:00-04:00',
    lastUpdate: '2025-07-07T09:20:00-04:00',
    priority: PRIORITY_TYPES.MEDIUM,
    progress: 100
  }
];

export const DONE_TASKS: TaskType[] = [
  {
    id: '15',
    title: 'Deploy staging environment',
    description:
      'Staging environment successfully deployed and tested with mock data.',
    status: STATUS_TYPES.DONE,
    createdAt: '2025-06-30T11:00:00-04:00',
    lastUpdate: '2025-07-03T15:30:00-04:00',
    priority: PRIORITY_TYPES.HIGH,
    progress: 100
  },
  {
    id: '16',
    title: 'Fix navbar responsiveness',
    description:
      'Navbar layout now works correctly across all screen sizes and breakpoints.',
    status: STATUS_TYPES.DONE,
    createdAt: '2025-07-01T10:00:00-04:00',
    lastUpdate: '2025-07-04T12:15:00-04:00',
    priority: PRIORITY_TYPES.MEDIUM,
    progress: 100
  },
  {
    id: '17',
    title: 'Set up CI/CD pipeline',
    description:
      'GitHub Actions configured to run tests and deploy to Vercel on push to main.',
    status: STATUS_TYPES.DONE,
    createdAt: '2025-07-02T09:30:00-04:00',
    lastUpdate: '2025-07-05T14:45:00-04:00',
    priority: PRIORITY_TYPES.HIGH,
    progress: 100
  },

  {
    id: '20',
    title: 'Test form validation logic',
    description:
      'Test cases written and validated for all form fields using React Hook Form.',
    status: STATUS_TYPES.DONE,
    createdAt: '2025-07-05T10:45:00-04:00',
    lastUpdate: '2025-07-08T11:30:00-04:00',
    priority: PRIORITY_TYPES.HIGH,
    progress: 100
  }
];
