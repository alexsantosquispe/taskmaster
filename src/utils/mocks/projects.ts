import type { ProjectDTO } from '../../services/apiTypes';

export const PROJECTS: ProjectDTO[] = [
  {
    id: '4e393d82-2ec9-46e6-89a1-f42aca499e24',
    name: 'Spotify Clone App',
    description: 'This is new project with react native',
    create_date: '2025-08-22T17:24:45.746846+00:00',
    color: '#2563eb',
    code: 'SCA',
    update_date: '2025-08-23T20:00:00.746846+00:00'
  },
  {
    id: 'ac2eed41-680b-4cd9-b079-7db1e4fac09e',
    name: 'Resume Builder V1',
    description:
      'Tihs project is a web app to create an resume and check the how it can be improved with AI',
    create_date: '2025-08-21T03:59:50.116667+00:00',
    color: '#f97316',
    code: 'RESB',
    update_date: '2025-08-21T03:59:50.116667+00:00'
  },
  {
    id: 'd9587eaa-1000-49db-8ce9-183fa47a1696',
    name: 'Task Master Mobile App',
    description: 'This is a mobile version with react native',
    create_date: '2025-08-21T03:40:21.137062+00:00',
    color: '#eab308',
    code: 'TMMA',
    update_date: '2025-08-21T03:40:21.137062+00:00'
  },
  {
    id: '5082b2a0-1e2a-43ca-9661-a12260520a15',
    name: 'Web Track App',
    description: 'Agile project management tool',
    create_date: '2025-08-18T23:51:48.391714+00:00',
    color: '#8B5CF6',
    code: 'DEV',
    update_date: '2025-08-18T23:51:48.391714+00:00'
  }
];
