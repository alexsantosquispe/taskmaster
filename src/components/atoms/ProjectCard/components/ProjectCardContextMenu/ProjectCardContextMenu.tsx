import { ContextMenu } from '@/components/atoms/ContextMenu/ContextMenu';
import { Modal } from '@/components/atoms/Modal/Modal';
import { PROJECT_MENU_OPTIONS } from '@/constants';
import { EllipsisVerticalIcon } from '@/icons';
import type { Option } from '@/models/types';
import { useState } from 'react';

export const DeleteProjectModal = ({ onClose }: { onClose: () => void }) => {
  return (
    <Modal
      title="Delete project"
      onClose={onClose}
      classNames={{ container: 'w-[40rem]' }}
    >
      <p>Are you sure you want to delete this project?</p>
    </Modal>
  );
};

export const EditProjectModal = ({ onClose }: { onClose: () => void }) => {
  return (
    <Modal
      title="Edit project"
      onClose={onClose}
      classNames={{ container: 'w-[40rem]' }}
    >
      <p>Edit project form goes here</p>
    </Modal>
  );
};

export const ProjectCardContextMenu = () => {
  const [modalType, setModalType] = useState<'edit' | 'delete' | null>(null);

  const handleSelectOption = (option: Option) => {
    if (option.value === 'edit') setModalType('edit');
    if (option.value === 'delete') setModalType('delete');
  };

  return (
    <>
      <ContextMenu
        ariaLabel="Project options"
        options={PROJECT_MENU_OPTIONS}
        onSelectOption={handleSelectOption}
        icon={<EllipsisVerticalIcon className="size-4" />}
        className={{
          mainContainer: '-mr-2',
          contentWrapper: 'min-w-[6rem]'
        }}
      />

      {modalType === 'delete' && (
        <DeleteProjectModal onClose={() => setModalType(null)} />
      )}
      {modalType === 'edit' && (
        <EditProjectModal onClose={() => setModalType(null)} />
      )}
    </>
  );
};
