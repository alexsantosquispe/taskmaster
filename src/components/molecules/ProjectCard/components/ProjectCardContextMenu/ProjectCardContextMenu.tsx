import { ContextMenu } from '@/components/atoms/ContextMenu/ContextMenu';
import { Modal } from '@/components/atoms/Modal/Modal';
import { DeleteProjectModal } from '@/components/molecules/DeleteProjectModal/DeleteProjectModal';
import { PROJECT_MENU_OPTIONS } from '@/constants';
import { EllipsisVerticalIcon } from '@/icons';
import type { Option } from '@/models/types';
import type { ProjectDTO } from '@/services/apiTypes';
import { useState } from 'react';

export const EditProjectModal = ({ onClose }: { onClose: () => void }) => {
  return (
    <Modal
      title="Edit project"
      onClose={onClose}
      classNames={{ container: 'w-[40rem]' }}
    >
      <p>Edit project form goes here</p>
    </Modal>
  );
};

interface ProjectCardContextMenuProps {
  projectItem: Omit<ProjectDTO, 'create_date' | 'update_date'>;
}

export const ProjectCardContextMenu = ({
  projectItem
}: ProjectCardContextMenuProps) => {
  const [modalType, setModalType] = useState<'edit' | 'delete' | null>(null);

  const handleSelectOption = (option: Option) => {
    if (option.value === 'edit') setModalType('edit');
    if (option.value === 'delete') setModalType('delete');
  };

  return (
    <>
      <ContextMenu
        ariaLabel="Project options"
        options={PROJECT_MENU_OPTIONS}
        onSelectOption={handleSelectOption}
        icon={<EllipsisVerticalIcon className="size-4" />}
        className={{
          mainContainer: '-mr-2',
          contentWrapper: 'min-w-[6rem]',
          button:
            'text-primary/80 hover:text-primary dark:text-white/70 dark:hover:text-white'
        }}
      />

      {modalType === 'delete' && (
        <DeleteProjectModal
          projectId={projectItem.id}
          projectName={projectItem.name}
          onClose={() => setModalType(null)}
        />
      )}

      {modalType === 'edit' && (
        <EditProjectModal onClose={() => setModalType(null)} />
      )}
    </>
  );
};
