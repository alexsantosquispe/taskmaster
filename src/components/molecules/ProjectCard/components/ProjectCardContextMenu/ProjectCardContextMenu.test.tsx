import '@testing-library/jest-dom';

import { fireEvent, render, screen } from '@testing-library/react';

import { PROJECT_MENU_OPTIONS } from '@/constants';
import { PROJECTS } from '@/utils/mocks/projects';
import { ProjectCardContextMenu } from './ProjectCardContextMenu';

describe('ProjectCardContextMenu', () => {
  const props = {
    projectItem: PROJECTS[0]
  };

  describe('styles', () => {
    it('should render the component correctly', () => {
      let component = render(<ProjectCardContextMenu {...props} />);

      expect(component).toMatchSnapshot();
    });
  });

  describe('behavior', () => {
    beforeEach(() => {
      render(<ProjectCardContextMenu {...props} />);
    });

    afterEach(() => {
      jest.clearAllMocks();
    });

    it('should open the context menu when click the button', () => {
      const optionsButton = screen.getByRole('button');

      fireEvent.click(optionsButton);

      expect(screen.getByTestId('context-menu')).toBeInTheDocument();

      expect(screen.getAllByRole('listitem')).toHaveLength(
        PROJECT_MENU_OPTIONS.length
      );
    });

    it('should hide the context menu when select an option and open the edit modal', () => {
      const optionsButton = screen.getByRole('button');

      fireEvent.click(optionsButton);

      expect(screen.getByTestId('context-menu')).toBeInTheDocument();

      const editOption = screen.getByText(PROJECT_MENU_OPTIONS[0].label);

      fireEvent.click(editOption);

      expect(screen.queryByTestId('context-menu')).not.toBeInTheDocument();

      expect(screen.getByTestId('modal')).toBeInTheDocument();

      expect(screen.getByText('Edit project')).toBeInTheDocument();

      const closeModalButton = screen.getByRole('button', {
        name: 'Close modal button'
      });

      expect(closeModalButton).toBeInTheDocument();

      fireEvent.click(closeModalButton);

      expect(screen.queryByTestId('modal')).not.toBeInTheDocument();
    });

    it('should hide the context menu when select an option and open the delete modal', () => {
      const optionsButton = screen.getByRole('button');

      fireEvent.click(optionsButton);

      expect(screen.getByTestId('context-menu')).toBeInTheDocument();

      const deleteOption = screen.getByText(PROJECT_MENU_OPTIONS[1].label);

      fireEvent.click(deleteOption);

      expect(screen.queryByTestId('context-menu')).not.toBeInTheDocument();

      expect(screen.getByTestId('modal')).toBeInTheDocument();

      expect(
        screen.getByText('Are you sure you want to delete this project?')
      ).toBeInTheDocument();

      const closeModalButton = screen.getByRole('button', {
        name: 'Close modal button'
      });

      expect(closeModalButton).toBeInTheDocument();

      fireEvent.click(closeModalButton);

      expect(screen.queryByTestId('modal')).not.toBeInTheDocument();
    });
  });
});
