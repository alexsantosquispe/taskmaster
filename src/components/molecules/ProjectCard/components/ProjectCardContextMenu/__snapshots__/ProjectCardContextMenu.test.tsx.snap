// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`ProjectCardContextMenu styles should render the component correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="relative flex flex-col -mr-2"
      >
        <button
          aria-label="Project options"
          class="flex w-fit items-center justify-center gap-2 text-sm font-semibold transition-colors duration-200 ease-in-out hover:cursor-pointer hover:bg-neutral-100 dark:hover:bg-white/15 rounded-md p-1 text-primary/80 hover:text-primary dark:text-white/70 dark:hover:text-white"
          type="button"
        >
          <div>
            <svg
              class="size-4"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <circle
                cx="12"
                cy="12"
                r="1"
              />
              <circle
                cx="12"
                cy="5"
                r="1"
              />
              <circle
                cx="12"
                cy="19"
                r="1"
              />
            </svg>
          </div>
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="relative flex flex-col -mr-2"
    >
      <button
        aria-label="Project options"
        class="flex w-fit items-center justify-center gap-2 text-sm font-semibold transition-colors duration-200 ease-in-out hover:cursor-pointer hover:bg-neutral-100 dark:hover:bg-white/15 rounded-md p-1 text-primary/80 hover:text-primary dark:text-white/70 dark:hover:text-white"
        type="button"
      >
        <div>
          <svg
            class="size-4"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="12"
              cy="12"
              r="1"
            />
            <circle
              cx="12"
              cy="5"
              r="1"
            />
            <circle
              cx="12"
              cy="19"
              r="1"
            />
          </svg>
        </div>
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
